scripts/azureml-assets/azureml/assets/release/model_publish_utils/utils.py

```diff
@@ -49,33 +42,40 @@ def _run(self, cmd, cwd: Path = "./") -> int:
             logger.print(f"Successfully executed! Output: \n{result.stdout}")
         return result.returncode
 
-    def _download_git_model(self) -> bool:
+    def _download_git_model(model_url, model_dir) -> bool:
         """Download the Model."""
         """
         Clones the model from the git URL into the Model Directory.
         Deletes the incomplete model artifact in case of failure.
-        Sets the HEAD to the commit hash.
         Deletes the .git folder in the downloaded artifacts.
         """
-        clone_cmd = f"git clone {self.model_url} {self.model_dir}"
-        result = self._run(clone_cmd)
+        clone_cmd = f"git clone {model_url} {model_dir}"
+        result = ModelDownloadUtils._run(clone_cmd)
         if result != 0:
             return False
-        if self.model_commit_hash:
-            cmd = f"git reset --hard {self.model_commit_hash}"
-            commit_exists = self._run(cmd, cwd=self.model_dir)
-            if commit_exists != 0:
-                # TODO: Error handling incase of incorrect commit hash.
-                logger.log_warning("Commit hash doesn't exist. Using model from latest HEAD.")
-        git_path = os.path.join(self.model_dir, '.git')
+        git_path = model_dir / ".git"
         shutil.rmtree(git_path, onerror=_onerror)
         return True
 
-    def download_model(self) -> bool:
+    def _download_azure_artifacts(model_url, model_dir) -> bool:
+        """Download model files from blobstore."""
+        commands = f"""
+        azcopy login --service-principal --application-id $SP_CLIENT_ID --tenant-id $SP_TENANT_ID
+        azcopy cp --recursive=true {model_url} {model_dir}
+        """
+        result = ModelDownloadUtils._run(commands)
+        # TODO: Handle error case correctly, since azcopy exits with 0 exit code, even in case of error.
```

Comment:
># TODO: Handle error case correctly, since azcopy exits with 0 exit code, even in case of error. [](http://example.com/codeflow?start=8&length=96)

Could we please create a separate gh issue & link it here?

Reply:
Created #283